examples/python/gui_demo.py

```diff
@@ -219,114 +248,93 @@ def create_tree_widget(self, parent_frame):
 
         popup = tk.Menu(tree, tearoff=0)
         popup.add_command(label="Remove")
-        popup.add_command(label="Rename")
+        # popup.add_command(label="Rename")
+        popup.add_command(label="Connect")
         self.tree_popup = popup
 
-        self.functions_node_id = 'root_function_blocks'
+    def tree_update(self, new_selected_node=None):
+        # make sure the tree is empty
+        self.tree.delete(*self.tree.get_children())
 
+        self.selected_node = new_selected_node
 
-    def create_property_widget(self, parent_frame):
-        frame = tk.Frame()
-        # define columns
-        tree = ttk.Treeview(frame, columns=('value', 'unit', 'access'), show='tree headings')
+        self.tree_traverse_components_recursive(self.instance)
+        self.properties_update()
+        self.attributes_update()
+        self.selected_node = self.tree_restore_selection(
+            self.selected_node)  # reset in case the selected node outdates
 
-        scroll_bar = ttk.Scrollbar(frame, orient="vertical", command=tree.yview)
-        tree.configure(yscrollcommand=scroll_bar.set)
-        scroll_bar.pack(side="right", fill="y")
-        tree.pack(fill="both", expand=True)
+    def tree_traverse_components_recursive(self, component):
+        self.tree_add_component(
+            '' if component.parent is None else component.parent.global_id, component)
+        if (daq.IFolder.can_cast_from(component)):
+            folder = daq.IFolder.cast_from(component)
+            for item in folder.items:
+                self.tree_traverse_components_recursive(item)
 
-        # define headings
-        tree.heading('#0', text='Property')
-        tree.heading('value', text='Value')
-        tree.heading('unit', text='Unit')
-        tree.heading('access', text='Access')
-        # layout
-        tree.column('#0', anchor=tk.CENTER)
-        tree.column('#1', anchor=tk.CENTER)
-        tree.column('#2', anchor=tk.CENTER, width=100, stretch=False)
-        tree.column('#3', anchor=tk.CENTER, width=150, stretch=False)
-        style = ttk.Style()
-        style.configure("Treeview.Heading", font='Arial 10 bold')
-        # bind double-click to editing
-        tree.bind('<Double-1>', self.set_tree_cell_value)
+    def tree_add_component(self, parent_node_id, component):
+        component_node_id = component.global_id
+        component_name = component.name
 
-        parent_frame.add(ttk.Label(parent_frame, text='Properties'))
-        parent_frame.add(frame, height=480)
-        self.property_tree = tree
+        if daq.IDevice.can_cast_from(component):
```

Comment:
Folder names at the moment use local ids. For now please check if component is IFolder and if it has the specified local_id, use fixed name:

"Sig": "Signals"
"FB": "Function blocks"
"Dev": "Devices"
"IP": "Input ports"
"IO": "Inputs/Outputs"